linkml/generators/sqltablegen.py

```diff
@@ -0,0 +1,217 @@
+    dialect: str
+    inject_primary_keys: bool = True
+
+    def __init__(self, schema: Union[str, TextIO, SchemaDefinition], dialect='sqlite',
+                 use_foreign_keys = True,
+                 rename_foreign_keys = False,
+                 direct_mapping = False,
+                 **kwargs) -> None:
+        # do not do generator-based transform of schema
+        #super().__init__(schema, **kwargs)
+        self.schema = schema
+        self.relative_slot_num = 0
+        self.dialect = dialect
+        self.use_foreign_keys = use_foreign_keys
+        self.rename_foreign_keys = rename_foreign_keys
+        self.direct_mapping = direct_mapping
+
+    def generate_ddl(self, naming_policy: SqlNamingPolicy = None, **kwargs) -> str:
+        ddl_str = ""
+        def dump(sql, *multiparams, **params):
+            nonlocal ddl_str
+            ddl_str += f"{str(sql.compile(dialect=engine.dialect)).rstrip()};"
+
+        engine = create_mock_engine(
+            f'{self.dialect}://./MyDb',
+            strategy='mock',
+            executor= dump)
+        schema_metadata = MetaData()
+        sqltr = RelationalModelTransformer(SchemaView(self.schema))
+        tr_result = sqltr.transform(**kwargs)
+        schema = tr_result.schema
+
+        def sql_name(n: str) -> str:
+            if naming_policy is not None:
+                if naming_policy == SqlNamingPolicy.underscore:
+                    return underscore(n)
+                elif naming_policy == SqlNamingPolicy.camelcase:
+                    return camelcase(n)
+                elif naming_policy == SqlNamingPolicy.preserve:
+                    return n
+                else:
+                    raise Exception(f'Unknown: {naming_policy}')
+            else:
+                return n
+
+        sv = SchemaView(schema)
+        for cn, c in schema.classes.items():
+            pk_slot = sv.get_identifier_slot(cn)
+            if c.attributes:
+                cols = []
+                for sn, s in c.attributes.items():
+                    is_pk = 'primary_key' in s.annotations
+                    if pk_slot:
+                        is_pk = sn == pk_slot.name
+                    #else:
+                    #    is_pk = True  ## TODO: use unique key
+                    args = []
+                    if s.range in schema.classes:
+                        fk = sql_name(self.get_foreign_key(s.range, sv))
+                        args = [ForeignKey(fk)]
+                    field_type = self.get_sql_range(s, schema)
+                    col = Column(sql_name(sn), field_type, *args, primary_key=is_pk, nullable=not s.required)
+                    cols.append(col)
+                Table(sql_name(cn), schema_metadata, *cols)
+        schema_metadata.create_all(engine)
+        return ddl_str
+
+    # TODO: merge with code from sqlddlgen
+    def get_sql_range(self, slot: SlotDefinition, schema: SchemaDefinition):
```

Comment:
Can we make `schema: SchemaDefinition = None`?

And add an if block in the body like such:

```
if not schema:
    schema = self.schema
```

Reply:
yes